redash/query_runner/pinot.py

```diff
@@ -0,0 +1,135 @@
+        logger.debug("Running query %s with username: %s, password: %s", query, self.username, self.password)
+        connection = pinotdb.connect(
+            host=self.configuration["brokerHost"],
+            port=self.configuration["brokerPort"],
+            path="/query/sql",
+            scheme=(self.configuration.get("brokerScheme") or "http"),
+            verify_ssl=False,
+            username=self.username,
+            password=self.password,
+        )
+
+        cursor = connection.cursor()
+
+        try:
+            cursor.execute(query)
+            logger.debug("cursor.schema = %s",cursor.schema)
+            columns = self.fetch_columns(
+                [(i["name"], PINOT_TYPES_MAPPING.get(i["type"], None)) for i in cursor.schema]
+            )
+            rows = [
+                dict(zip((column["name"] for column in columns), row)) for row in cursor
+            ]
+
+            data = {"columns": columns, "rows": rows}
+            error = None
+            json_data = json_dumps(data)
+            logger.debug("Pinot execute query [%s], and resp: %s", query, json_data)
+        finally:
+            connection.close()
+
+        return json_data, error
+
+    def get_schema(self, get_stats=False):
+        schema = {}
+        for schema_name in self.get_schema_names():
+            for table_name in self.get_table_names():
+                schema_table_name = "{}.{}".format(schema_name, table_name)
+                if table_name not in schema:
+                    schema[schema_table_name] = {"name": schema_table_name, "columns": []}
+                table_schema =self.get_pinot_table_schema(table_name)
+
+                for column in table_schema.get("dimensionFieldSpecs", []) + table_schema.get(
+                    "metricFieldSpecs", []) + table_schema.get("dateTimeFieldSpecs", []):
+                    c = {
+                            "name": column["name"],
+                            "type": PINOT_TYPES_MAPPING[column["dataType"]],
+                        }
+                    schema[schema_table_name]["columns"].append(c)
+        return list(schema.values())
+
+    def get_schema_names(self):
+        return ["default"]
+
+    def get_pinot_table_schema(self, pinot_table_name):
+        return self.get_metadata_from_controller("/tables/" + pinot_table_name + "/schema")
+
+    def get_table_names(self):
+        return self.get_metadata_from_controller("/tables")["tables"]
+
+    def get_metadata_from_controller(self, path):
+        url = self.controller_uri + path
+        r = requests.get(url, headers={"Accept": "application/json"}, auth= HTTPBasicAuth(self.username, self.password))
+        try:
+            result = r.json()
+            logger.debug("get_metadata_from_controller from path %s: %s", path, result)
+        except ValueError as e:
+            raise pinotdb.exceptions.DatabaseError(
+                "Got invalid json response from " f"{self.controller_uri}:{path}: {r.text}"
```

Comment:
```suggestion
                f"Got invalid json response from {self.controller_uri}:{path}: {r.text}"
```

I think the original version of this is invalid Python.